modules/optimization/include/optimizationreporters/OptimizationReporter.h

```diff
@@ -0,0 +1,101 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "OptimizationData.h"
+#include "DataIO.h"
+#include "libmesh/petsc_vector.h"
+#include "libmesh/petsc_matrix.h"
+// friends
+#include "OptimizeSolve.h"
+#include "OptimizationReporterTest.h"
+
+class OptimizationReporter : public OptimizationData
+{
+public:
+  static InputParameters validParams();
+  OptimizationReporter(const InputParameters & parameters);
+
+  void initialize() override final {}
+  void execute() override final {}
+  void finalize() override final {}
+
+  /**
+   * Function to initialize petsc vectors from vpp data
+   * FIXME: this should be const
+   */
+  void setInitialCondition(libMesh::PetscVector<Number> & param);
+
+  /**
+   * Function to over-ride misfit values with the simulated values from the matrix free hessian
+   * forward solve
+   */
+  void setMisfitToSimulatedValues();
+
+  /**
+   * Functions to get and check bounds
+   */
+  bool hasBounds() const { return _upper_bounds.size() > 0 && _lower_bounds.size() > 0; }
+  const std::vector<Real> & getUpperBounds() const { return _upper_bounds; };
+  const std::vector<Real> & getLowerBounds() const { return _lower_bounds; };
+  /**
+   * Function to compute default bounds when user did not provide bounds
```

Comment:
docstrings
I need to know why it s a vector for example

Reply:
done

Reply:
add @return field please